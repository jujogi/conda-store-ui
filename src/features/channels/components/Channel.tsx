import SquareIcon from "@mui/icons-material/Square";
import Box from "@mui/material/Box";
import Typography from "@mui/material/Typography";
import useTheme from "@mui/material/styles/useTheme";
import React from "react";

interface IChannelProps {
  /**
   * TODO: this interface needs a docstring for each param
   */
  channelName: string;
}

export const Channel = ({ channelName }: IChannelProps) => {
  const { palette } = useTheme();

  return (
    <Box className="box" sx={{ display: "flex", alignItems: "center" }}>
      <SquareIcon
        sx={{
          color: palette.primary.main,
          width: 10,
          height: 10,
          marginRight: "12px"
        }}
      />
      <Typography className="typography" sx={{ color: "#4D4D4D" }}>
        {channelName}
      </Typography>
    </Box>
  );
};
