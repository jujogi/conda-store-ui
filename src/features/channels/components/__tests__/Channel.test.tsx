import React from "react";
import { getByTestId, render, screen, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import "@testing-library/jest-dom/extend-expect";
import { Channel } from "../Channel";
import ReactDOM from "react-dom";
import { create } from "react-test-renderer";

test("render Box component", () => {
  const { container } = render(<Channel channelName="test" />);
  const box = container.getElementsByClassName("box");
  expect(box).toBeTruthy();

  const typography = container.getElementsByClassName("typography");
  expect(typography).toBeTruthy();

  expect(
    getByTestId(document.documentElement, "SquareIcon")
  ).toBeInTheDocument();

  expect(screen.getByTestId("SquareIcon")).toHaveStyle({
    marginRight: "12px",
    height: "10px",
    width: "10px",
    color: "rgb(25, 118, 210)"
  });
});

//Snapshot
// const tree = create(<Channel channelName="test" />);
// test("snapshot", () => {
//   expect(tree).toMatchSnapshot();
// });
