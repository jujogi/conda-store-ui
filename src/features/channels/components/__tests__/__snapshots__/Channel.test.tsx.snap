// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="MuiBox-root css-70qvj9"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1fevu13-MuiSvgIcon-root"
    data-testid="SquareIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M3 3h18v18H3z"
    />
  </svg>
  <p
    className="MuiTypography-root MuiTypography-body1 box1 css-14nmori-MuiTypography-root"
  >
    test
  </p>
</div>
`;
